File: Perl/Problem2.pl
#! /usr/bin/perl
use strict;
use warnings;

my $sum = 0;
my $fib1 = 1;
my $fib2 = 1;
my $fibNext = 0;

while ($fibNext <= 4000000) {
      $fibNext = $fib1 + $fib2;
      $fib1 = $fib2;
      $fib2 = $fibNext;
      if (!($fibNext % 2)) {
         $sum += $fibNext;
      }
}

print "$sum\n";

File: Python/Problem2.py
#! /usr/local/bin/python3.1

sum = 0
fib1 = 1
fib2 = 2

while fib1 < 4000000:
    if ((fib1%2) == 0):
        sum += fib1
    fib1, fib2 = fib2, fib1 + fib2
print(sum)
